tests/Adapters/Microsoft.Bot.Builder.Adapters.Facebook.SecondaryTestBot/Bots/EchoBot.cs

```diff
@@ -0,0 +1,100 @@
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Adapters.Facebook.SecondaryTestBot.Bots
+{
+    public class EchoBot : ActivityHandler
+    {
+        /// <summary>
+        /// This option passes thread control from the secondary receiver to a primary receiver.
+        /// </summary>
+        private const string OptionPassPrimaryBot = "Pass to primary";
+
+        protected override async Task OnMembersAddedAsync(IList<ChannelAccount> membersAdded, ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var activity = MessageFactory.Text("Hello and Welcome!");
+            await turnContext.SendActivityAsync(activity, cancellationToken);
+        }
+
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if (turnContext.Activity.Attachments != null)
+            {
+                foreach (var attachment in turnContext.Activity.Attachments)
+                {
+                    var activity = MessageFactory.Text($" I got {turnContext.Activity.Attachments.Count} attachments");
+
+                    var image = new Attachment(
+                       attachment.ContentType,
+                       content: attachment.Content);
+
+                    activity.Attachments.Add(image);
+                    await turnContext.SendActivityAsync(activity, cancellationToken);
+                }
+            }
+            else
+            {
+                IActivity activity;
+
+                switch (turnContext.Activity.Text)
+                {
+                    case OptionPassPrimaryBot:
+                        activity = MessageFactory.Text("Redirecting to the primary bot...");
+                        activity.Type = ActivityTypes.Event;
+                        (activity as IEventActivity).Name = "pass_thread_control";
+                        (activity as IEventActivity).Value = "An app id of the primary receiver";
+                        break;
+
+                    case "Redirected to the bot":
+                        activity = MessageFactory.Text("Hello Human, I'm the secondary bot to help you!");
+                        break;
+                    default:
+                        activity = MessageFactory.Text($"Echo Secondary: {turnContext.Activity.Text}");
+                        break;
+                }
+
+                await turnContext.SendActivityAsync(activity, cancellationToken);
+            }
+        }
+
+        protected override async Task OnEventActivityAsync(ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if (turnContext.Activity.Value != null)
+            {
+                var activity = MessageFactory.Text("Hello Again Human, I'm the bot the secondary bot to help you!");
```

Comment:
Is this dialog correct?

Reply:
nope. But we will fix this in an later pass.